src/classes/Collision/SAT.js

```diff
@@ -0,0 +1,187 @@
+Goblin.Collision.SAT = {
+    /**
+     * Performs A SAT collision between to objects. Returns projection (that can be used as a minimum translation vector) or null if there is no collision.
+     * @param {Goblin.RigidBody|Goblin.RigidBodyProxy} objectA
+     * @param {Goblin.RigidBody|Goblin.RigidBodyProxy} objectB
+     * @returns {Goblin.Collision.SAT.Projection|null}
+     */
+    performSat: ( function() {
+        var negatedNormal = new Goblin.Vector3();
+
+        return function( objectA, objectB ) {
+            var allNormals = [];
+            var normalsA = objectA.faceNormals;
+            var normalsB = objectB.faceNormals;
+            if ( normalsA.length === 0 ) {
+                allNormals = normalsB;
+            } else if ( normalsB === 0 ) {
+                allNormals = normalsA;
+            } else {
+                var crossNormals = this._getAllCrossProductPairs( normalsA, normalsB );
+                allNormals.push.apply( allNormals, normalsA );
+                allNormals.push.apply( allNormals, normalsB );
+                allNormals.push.apply( allNormals, crossNormals );
+                this.removeDuplicatedVectors( allNormals );
+            }
+
+            var allProjections = [];
+            var minimumProjection = null;
+            var minimumOverlap = Infinity;
+
+            for ( var i = 0; i < allNormals.length; i++ ) {
```

Comment:
This is a nice, but quite a complex method. Could you please add inline comments to explain what's going on?

Reply:
Done. I've also simplified SAT a little bit, because I'm not sure how it's api will look like after hull vs hull collision